cvmfs/download.h

```diff
@@ -138,6 +145,11 @@ struct JobInfo {
     probe_hosts = false;
     head_request = false;
     follow_redirects = false;
+    pid = -1;
```

Comment:
I'd rather not patch curl, since it prevents folks from using the system version.

*However*, in follow-up work, I did find that curl has a nasty habit of caching HTTP connections and SSL sessions (even if the file changes!).  One performance improvement we can tackle later is see if we can manage session reuse more explicitly to avoid disabling caches.